src/daft-core/src/array/ops/utf8.rs

```diff
@@ -60,6 +63,39 @@ fn is_valid_input_lengths(lengths: &[usize]) -> bool {
     }
 }
 
+/// Parse inputs for string operations.
+/// Returns a tuple of (is_full_null, expected_size).
+fn parse_inputs<T>(
+    self_arr: &Utf8Array,
+    other_arrs: &[&DataArray<T>],
+) -> Result<(bool, usize), String>
+where
+    T: DaftPhysicalType,
+{
+    assert!(!other_arrs.is_empty());
```

Comment:
Does this need to be a strong assertion? Can't we return according to just `self_arr` in this case?

Reply:
yes, makes sense.